numpy/_core/tests/test_numeric.py

```diff
@@ -1039,20 +1039,26 @@ def check_promotion_cases(self, promote_func):
         assert_equal(promote_func(np.array([b]), u8), np.dtype(np.uint8))
         assert_equal(promote_func(np.array([b]), i32), np.dtype(np.int32))
         assert_equal(promote_func(np.array([b]), u32), np.dtype(np.uint32))
-        assert_equal(promote_func(np.array([i8]), i64), np.dtype(np.int8))
-        assert_equal(promote_func(u64, np.array([i32])), np.dtype(np.int32))
-        assert_equal(promote_func(i64, np.array([u32])), np.dtype(np.uint32))
+        assert_equal(promote_func(np.array([i8]), i64), np.dtype(np.int64))
+        # unsigned and signed unfortunately tend to promote to float64:
+        assert_equal(promote_func(u64, np.array([i32])), np.dtype(np.float64))
```

Comment:
Sorry for coming into this super late - this seems like a surprising corner case. I guess this was discussed before though?

Reply:
The PR doesn't change the behavior at all, it just before this was the same as `python int == int32`, the mixing was always strange like that.